### PR TITLE
feat: added support notice for flex feeds

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

This PR introduces an informational notice displayed at the top of the validation report when GTFS Flex feeds are detected.

**Expected Behavior:**

When a Flex feed is detected, the following informational message is shown:

image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by emmambd** [comment]
Tested that this displays for any feed with Flex features and -doesn't- display for feeds without Flex. This is great! 

A veryyy small nit but remove "the" in the second sentence, so it's: "Please note that GTFS Flex validation support in still in development". Then you're good to go :)

File: main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java [hunk 1]
```diff
@@ -591,4 +591,13 @@ public void setFilenames(ImmutableSortedSet<String> filenames) {
...
     return filenames;
   }
+
+  public Boolean hasFlexFeatures() {
+    return specFeatures.keySet().stream()
+        .anyMatch(
+            feature ->
+                feature.getFeatureGroup() != null
+                    && feature.getFeatureGroup().equals("Flexible Services")
+                    && specFeatures.get(feature));
```
Comment (jcpitre) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java:601]: Out of scope for this PR, but I wish specFeatures name was changed to specFeaturesPresence or something like this. The current name is misleading.

**Review by jcpitre — APPROVED** [review]
LGTM

**Comment by emmambd** [comment]
This notice appears for feeds with just the `Continuous Stops` feature, which isn't a GTFS-Flex service even though it's in the Flexible Services group. We should modify it to display for every feature in the Flexible Services group except `Continuous Stops`.

**Pull request merged**